File: optigold_mobile/lib/core/api/models/market_snapshot.dart
class MarketSnapshot {
  final DateTime capturedAt;
  final double? gldPrice;
  final double? gld24hChangePct;
  final double? gldAtmIv;
  final double? gldIvRank;
  final double? gldIvPercentile;
  final double? gldHv20d;
  final double? vixLevel;
  final double? ivPremiumRatio;
  final double? putCallSkew;
  final String? marketRegime;
  final double? regimeConfidence;
  final double? ema20;
  final double? ema50;
  final double? rsi14;
  final double? adx;
  final String? marketSession;

  const MarketSnapshot({
    required this.capturedAt,
    this.gldPrice,
    this.gld24hChangePct,
    this.gldAtmIv,
    this.gldIvRank,
    this.gldIvPercentile,
    this.gldHv20d,
    this.vixLevel,
    this.ivPremiumRatio,
    this.putCallSkew,
    this.marketRegime,
    this.regimeConfidence,
    this.ema20,
    this.ema50,
    this.rsi14,
    this.adx,
    this.marketSession,
  });

  factory MarketSnapshot.fromJson(Map<String, dynamic> j) => MarketSnapshot(
        capturedAt:       DateTime.parse(j['captured_at'] as String),
        gldPrice:         (j['gld_price'] as num?)?.toDouble(),
        gld24hChangePct:  (j['gld_24h_change_pct'] as num?)?.toDouble(),
        gldAtmIv:         (j['gld_atm_iv'] as num?)?.toDouble(),
        gldIvRank:        (j['gld_iv_rank'] as num?)?.toDouble(),
        gldIvPercentile:  (j['gld_iv_percentile'] as num?)?.toDouble(),
        gldHv20d:         (j['gld_hv_20d'] as num?)?.toDouble(),
        vixLevel:         (j['vix_level'] as num?)?.toDouble(),
        ivPremiumRatio:   (j['iv_premium_ratio'] as num?)?.toDouble(),
        putCallSkew:      (j['put_call_skew'] as num?)?.toDouble(),
        marketRegime:     j['market_regime'] as String?,
        regimeConfidence: (j['regime_confidence'] as num?)?.toDouble(),
        ema20:            (j['ema20'] as num?)?.toDouble(),
        ema50:            (j['ema50'] as num?)?.toDouble(),
        rsi14:            (j['rsi_14'] as num?)?.toDouble(),
        adx:              (j['adx'] as num?)?.toDouble(),
        marketSession:    j['market_session'] as String?,
      );

  bool get isBullish => marketRegime?.contains('bull') == true;
  bool get isBearish => marketRegime?.contains('bear') == true;

  String get formattedRegime {
    if (marketRegime == null) return 'Unknown';
    return marketRegime!.replaceAll('_', ' ').split(' ')
        .map((w) => w[0].toUpperCase() + w.substring(1))
        .join(' ');
  }
}

File: backend/app/models/db/market_snapshot.py
from datetime import datetime
from typing import Optional
from sqlalchemy import Float, Integer, String, DateTime
from sqlalchemy.orm import Mapped, mapped_column
from app.models.db.base import Base


class MarketSnapshot(Base):
    __tablename__ = "market_snapshot"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    captured_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    gld_price: Mapped[Optional[float]] = mapped_column(Float)
    gld_24h_change_pct: Mapped[Optional[float]] = mapped_column(Float)
    gld_atm_iv: Mapped[Optional[float]] = mapped_column(Float)
    gld_iv_rank: Mapped[Optional[float]] = mapped_column(Float)
    gld_iv_percentile: Mapped[Optional[float]] = mapped_column(Float)
    gld_hv_20d: Mapped[Optional[float]] = mapped_column(Float)
    vix_level: Mapped[Optional[float]] = mapped_column(Float)
    iv_premium_ratio: Mapped[Optional[float]] = mapped_column(Float)
    put_call_skew: Mapped[Optional[float]] = mapped_column(Float)
    market_regime: Mapped[Optional[str]] = mapped_column(String(30))
    regime_confidence: Mapped[Optional[float]] = mapped_column(Float)
    ema20: Mapped[Optional[float]] = mapped_column(Float)
    ema50: Mapped[Optional[float]] = mapped_column(Float)
    rsi_14: Mapped[Optional[float]] = mapped_column(Float)
    adx: Mapped[Optional[float]] = mapped_column(Float)
    market_session: Mapped[Optional[str]] = mapped_column(String(20))
